File: VB/GridDataAttributes/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Views.Grid

Namespace GridDataAttributes

    Public Partial Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
            gridView1.OptionsView.ShowGroupPanel = False
            AddHandler gridControl1.DataSourceChanged, AddressOf gridControl1_DataSourceChanged
        End Sub

        Private Sub gridControl1_DataSourceChanged(ByVal sender As Object, ByVal e As EventArgs)
            gridControl1.MainView.PopulateColumns()
            TryCast(gridControl1.MainView, GridView).BestFitColumns()
        End Sub

        Protected Overrides Sub OnLoad(ByVal e As EventArgs)
            MyBase.OnLoad(e)
        'gridControl1.DataSource = CompanyProductList.GetSampleData();
        'companyProductListBindingSource.DataSource = CompanyProductList.GetSampleData();
        'companyProductBindingSource.DataSource = CompanyProductList.GetSampleData();
        End Sub

        Private Sub barEditItem3_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            Select Case TryCast(barEditItem3.EditValue, String)
                Case "Company public info"
                    gridControl1.DataSource = GridSampleDataList.GetCompanyPublicInfo()
                Case "Company private info"
                    gridControl1.DataSource = GridSampleDataList.GetCompanyPrivateInfo()
                Case "Product info"
                    gridControl1.DataSource = GridSampleDataList.GetProductSample()
            End Select
        End Sub
    End Class
End Namespace

File: CS/GridDataAttributes/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid.Views.Grid;

namespace GridDataAttributes {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            gridView1.OptionsView.ShowGroupPanel = false;
            gridControl1.DataSourceChanged += gridControl1_DataSourceChanged;
        }

        void gridControl1_DataSourceChanged(object sender, EventArgs e) {
            gridControl1.MainView.PopulateColumns();
            (gridControl1.MainView as GridView).BestFitColumns();
        }

        protected override void OnLoad(EventArgs e) {
            base.OnLoad(e);
            //gridControl1.DataSource = CompanyProductList.GetSampleData();
            //companyProductListBindingSource.DataSource = CompanyProductList.GetSampleData();
            //companyProductBindingSource.DataSource = CompanyProductList.GetSampleData();
        }

        private void barEditItem3_EditValueChanged(object sender, EventArgs e) {
            switch (barEditItem3.EditValue as string) {
                case "Company public info":
                    gridControl1.DataSource = GridSampleDataList.GetCompanyPublicInfo();
                    break;
                case "Company private info":
                    gridControl1.DataSource = GridSampleDataList.GetCompanyPrivateInfo();
                    break;
                case "Product info":
                    gridControl1.DataSource = GridSampleDataList.GetProductSample();
                    break;
            }
        }
    }
}
